clock/test_clock_test.go

```diff
@@ -61,3 +63,39 @@ func TestTickAfter(t *testing.T) {
 	tickOrTimeOut(ticker2, true)
 	tickOrTimeOut(ticker3, false)
 }
+
+// TestTickSignal tests that TickAfter signals registration allowing
+// safe time advancement.
+func TestTickSignal(t *testing.T) {
+	ch := make(chan struct{})
+	c := NewTestClockWithTickSignal(testTime, ch)
+
+	var wg sync.WaitGroup
+	timeout := make(chan struct{}, 1)
```

Comment:
If you make this an error channel, you don't need the wait group

Reply:
ptal